### PR TITLE
Fix Pyodide REPL URL

### PR DESCRIPTION
**Pull request by lesteve**
### Description

The doc about [accessing Python scope from Javascript](
https://pyodide.org/en/latest/usage/quickstart.html#accessing-python-scope-from-javascript)

image

mentions the URL https://cdn.jsdelivr.net/pyodide/v0.21.3/full/console.html which ends up on page showing only "Forbidden"
image

I am guessing this was meant to be pointing to a Pyodide REPL URL instead. I used the `latest` Pyodide REPL URL rather than the `stable` one because this is what is done at the top of the page (both for the `latest` and the `stable` doc).

### Checklists

The following don't really apply to this PR so I checked them all.

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests
- [x] Add new / update outdated documentation

### CODE REVIEW

**Comment by hoodmane** [comment]
Thanks @lesteve!

Once upon a time `PYODIDE_CDN_URL` pointed to `iodide.io` but mozilla stopped hosting that. By the way @rth I suppose that we should try to acquire the old iodide.io domain because there is some danger someone could camp it and put something evil at old links.

This isn't quite right though: we still need to template the version because we don't want to link the unstable Pyodide from stable versions of the docs.

**Comment by rth** [comment]
> I suppose that we should try to acquire the old iodide.io domain

It's set to expire in 2023 and I'm not sure if Mozilla won't renew it, even if it's discontinued, exactly to avoid that situation.

**Comment by rth** [comment]
Yes, the dev REPL is https://pyodide.org/en/latest/console.html and the latest stable releases is https://pyodide.org/en/stable/console.html . Tagged stable versions are also accessible e.g. https://pyodide.org/en/0.21.3/console.html but not alpha releases (e.g. 0.22.0a2) as we haven't yet figured out how to auto-register those on ReadtheDocs from tags.

Actually instead of templating, we could probably just use a relative path to ./console.hml in reathedocs so it would take the current one?

**Comment by lesteve** [comment]
> Actually instead of templating, we could probably just use a relative path to ./console.hml in reathedocs so it would take the current one?

I tried this in my last commit

**Comment by lesteve** [comment]
Does not seem like it is working in `.md` files ... probably for some obscure MyST reason ... 

There is no link generated here:
https://pyodide--3244.org.readthedocs.build/en/3244/usage/quickstart.html
image

I replaced them with HTML, let's see

File: docs/usage/quickstart.md [hunk 1]
```diff
@@ -142,13 +142,9 @@ you can access the global variable `x` from JavaScript in your browser's
...
 
-You can try it yourself in the browser console. Go to
-
-```text
-{{PYODIDE_CDN_URL}}console.html
-```
-
-and type the following into the browser console:
+You can try it yourself in the browser console. Go to the [Pyodide REPL
+URL](./console.html) and type the following into
```
[the remark this answers is not in the document]
Comment (lesteve) [reply docs/usage/quickstart.md]: Ah that would make a lot of sense indeed!

**Comment by lesteve** [comment]
Nope still not the case I am going to build the doc locally to debug this in a more efficient manner.

**Comment by rth** [comment]
Thanks! FYI you can also add `[skip ci]` to the commit message to skip build CI, the readthedocs CI is still going to run then.

**Comment by rth** [comment]
Also if it helps the readthedocs build log should be in https://readthedocs.org/projects/pyodide/builds/

**Comment by lesteve** [comment]
It seems to work now. The quickest work-around I found was to write some rst to avoid myst_parser cleverness ...

Wild-guess: it seems like sphinx tries to resolve the link but maybe ../console.html does not exist when building the doc?

Even when using HTML links, rather than markdown syntax, it does not work, probably because of myst_parser cleverness parsing the HTML link?

Possibly better work-arounds:
- create console.html in the sphinx source dir so that myst_parser can resolve it (haven't tried but may work)
- look at https://myst-parser.readthedocs.io/en/latest/develop/_changelog.html#markdown-link-resolution-improvements and see whether there are ways to configure so that it does not try to resolve `../console.html` (i.e. treats it as an external link)

**Comment by lesteve** [comment]
Alternatively I found that using a HTML link and setting class="external" does work too, e.g. something like this:
```html
<a class="reference external" href="../console.html">attempt with external</a>
```

(I also used "reference" class because that to be consistent with the other external links generated from markdown syntax)

**Review by rth — APPROVED** [review]
Thanks for investigating this!

Let's keep the rst even if certainly not ideal. The HTML class seems very magic and I guess there is no guarantee that it won't stop working at some point.

**Comment by hoodmane** [comment]
This needs backport.

**Pull request merged**